pmlc/rt/prng.cc

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Intel Corporation
+#include "mlir/ExecutionEngine/RunnerUtils.h"
+#include "pmlc/compiler/registry.h"
+#include "pmlc/util/logging.h"
+
+extern "C" void
+plaidml_rt_prng(unsigned stateRank, UnrankedMemRefType<float> *state,
+                unsigned resultRank, UnrankedMemRefType<float> *result,
+                unsigned newStateRank, UnrankedMemRefType<float> *newState) {
```

Comment:
These actually aren't `UnrankedMemRefType<float>*`, they are actually `StridedMemRefType<T, N>*`.

So I would replace this with:

```
plaidml_rt_prng(unsigned stateRank, StridedMemRefType<int32_t, 1> *state,
                unsigned resultRank, StridedMemRefType<float, 1> *result,
                unsigned newStateRank, StridedMemRefType<int32_t, 1> *newState) {
```

And then you can drop the casts below.

Reply:
Resolved